astropy/units/mixins.py

```diff
@@ -0,0 +1,33 @@
+import numpy as np
+import astropy.units as u
+
+__all__ = [
+    "QuantityOperatorsMixin",
+]
+
+
+class QuantityOperatorsMixin(np.lib.mixins.NDArrayOperatorsMixin):
+
+    def __mul__(self, other):
+        if isinstance(other, (u.UnitBase, str)):
```

Comment:
Ah, I guess I now see a bit that this suggestion also has a problem. What I was thinking about is something more like
```
       if isinstance(other, (u.UnitBase, str)):
           try:
               other = u.Unit(other)
           except Exception:
               return NotImplemented
           else:
               return self.__class__(self.value, self.unit * other)
        else:
            return super().__mul__(other)
```
But obviously, this makes an assumption about how the class is initialized...

Reply:
I don't see what the problem is, can you elaborate a little further?

Reply:
Sorry, that was not very clear; I wrote that after seeing your comment that `np.left_shift` doesn't work - and here I realized why `Quantity` was important at all. As I noted, my idea for this class was not to use `__array_ufunc__` at all, but just allow one to capture the cases where interaction with a `Unit` makes sense.